source/slang/slang-check-overload.cpp

```diff
@@ -42,20 +57,26 @@ namespace Slang
         {
             if (auto typeParam = as<GenericTypeParamDecl>(m))
             {
-                counts.allowed++;
+                if (counts.allowed >= 0)
+                    counts.allowed++;
                 if (!typeParam->initType.Ptr())
                 {
                     counts.required++;
                 }
             }
             else if (auto valParam = as<GenericValueParamDecl>(m))
             {
-                counts.allowed++;
+                if (counts.allowed >= 0)
+                    counts.allowed++;
                 if (!valParam->initExpr)
                 {
                     counts.required++;
                 }
             }
+            else if (as<GenericTypePackParamDecl>(m))
+            {
+                counts.allowed = -1;
```

Comment:
I noticed we more than once use a `-1` as a special parameter count state.
I wonder if we should make this an unscoped enum (or something more distinct) so it would be easier to refactor/read this code and make sense of what the -1 implies in the future.